crates/hyperswitch_connectors/src/connectors/redsys/transformers.rs

```diff
@@ -1,228 +1,1499 @@
+        three_ds_method_url: three_d_s_method_u_r_l,
+        three_ds_method_data,
+        message_version: protocol_version.clone(),
+        directory_server_id: three_d_s_server_trans_i_d,
+        three_ds_method_data_submission: true,
+        next_action_url,
+    };
+
+    let connector_metadata = Some(
+        serde_json::to_value(&three_ds_data)
+            .change_context(errors::ConnectorError::RequestEncodingFailed)
+            .attach_printable("Failed to serialize ThreeDsData")?,
+    );
+
+    Ok(RouterData {
+        status: enums::AttemptStatus::AuthenticationPending,
+        response: Ok(PaymentsResponseData::TransactionResponse {
+            resource_id: ResponseId::ConnectorTransactionId(response_data.ds_order.clone()),
+            redirection_data: Box::new(None),
+            mandate_reference: Box::new(None),
+            connector_metadata,
+            network_txn_id: None,
+            connector_response_reference_id: Some(response_data.ds_order.clone()),
+            incremental_authorization_allowed: None,
+            charges: None,
+        }),
+        ..item.data
+    })
+}
+
+fn handle_threeds_invoke_exempt<F>(
+    item: ResponseRouterData<F, RedsysResponse, PaymentsPreProcessingData, PaymentsResponseData>,
+    response_data: &RedsysPaymentsResponse,
+    three_d_s_server_trans_i_d: String,
+    protocol_version: String,
+) -> Result<
+    RouterData<F, PaymentsPreProcessingData, PaymentsResponseData>,
+    error_stack::Report<errors::ConnectorError>,
+> {
+    let three_ds_data = ThreeDsInvokeExempt {
+        message_version: protocol_version.clone(),
+        three_d_s_server_trans_i_d,
+    };
+
+    let connector_metadata = Some(
+        serde_json::to_value(&three_ds_data)
+            .change_context(errors::ConnectorError::RequestEncodingFailed)
+            .attach_printable("Failed to serialize ThreeDsData")?,
+    );
+
+    Ok(RouterData {
+        status: enums::AttemptStatus::AuthenticationPending,
+        response: Ok(PaymentsResponseData::TransactionResponse {
+            resource_id: ResponseId::ConnectorTransactionId(response_data.ds_order.clone()),
+            redirection_data: Box::new(None),
+            mandate_reference: Box::new(None),
+            connector_metadata,
+            network_txn_id: None,
+            connector_response_reference_id: Some(response_data.ds_order.clone()),
+            incremental_authorization_allowed: None,
+            charges: None,
+        }),
+        ..item.data
+    })
+}
+
+pub const SIGNATURE_VERSION: &str = "HMAC_SHA256_V1";
+
+fn base64_decode(input: &str) -> Result<Vec<u8>, error_stack::Report<errors::ConnectorError>> {
+    STANDARD
+        .decode(input)
+        .change_context(errors::ConnectorError::RequestEncodingFailed)
+        .attach_printable("Base64 decoding failed")
+}
+
+fn des_encrypt(
+    message: &str,
+    key: &str,
+) -> Result<String, error_stack::Report<errors::ConnectorError>> {
+    let iv_array = [0u8; 8];
```

Comment:
Add comments as to why we're using a zero IV.
If a link to the public documentation is available, we can include it as well.